### ISSUE TITLE
"<scalar> <filter-op> <vector>" takes the sample value from the wrong side

### ISSUE THREAD

**Issue by juliusv** [VictoriaMetrics/VictoriaMetrics#673]
**Describe the bug**
In native PromQL, binary filter operators between a scalar and a vector (e.g. `1 < foo`) always return the sample value from the vector side of the operation, no matter which side of the operator it is on. VictoriaMetrics always returns the sample value from the left-hand side, even if it is the scalar value.

**Version**

```
$ ./victoria-metrics-prod --version
victoria-metrics-20200727-204526-tags-v1.39.1-0-g5b81bdde3
```

**Comment by valyala** [comment]
@juliusv , thanks for filing the issue!

The issue must be fixed in the commit 68e4f40a72b13ac05baba4a519365a70b05be86b , which will be included in the next release of VictoriaMetrics .

**Comment by valyala** [comment]
The bugfix has been included in [v1.39.4](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the bug as fixed.
